homework/index.js

```diff
@@ -29,14 +29,116 @@
+            if (a.toLowerCase() < b.toLowerCase()) return -1;
+            if (a.toLowerCase() > b.toLowerCase()) return 1;
+            return 0;
+          });
+
+        const header = createAndAppend('header', root, { id: 'header' });
+        const headerDiv = createAndAppend('div', header, { id: 'headerDiv' });
+        createAndAppend('span', headerDiv, {
+          text: 'HYF Repositories',
+          id: 'headerSpan',
+        });
+        const select = createAndAppend('select', headerDiv, { id: 'selectId' });
+        addOptions(select, repoNames);
+        // to get the text(repo. name) from select element to be used to retrieve info about the selected repo
+        let selectedOption = document.getElementById('selectId').options[
+          document.getElementById('selectId').selectedIndex
+        ].text;
+        const repoDetailsDiv = createAndAppend('div', root, { id: 'repoDetailsDiv' });
+        const repoDiv = createAndAppend('div', repoDetailsDiv, { class: 'divClass' });
+        const descriptionDiv = createAndAppend('div', repoDetailsDiv, {
+          class: 'divClass',
+          id: 'descDiv',
+        });
+        const forkDiv = createAndAppend('div', repoDetailsDiv, { class: 'divClass' });
+        const updatedDiv = createAndAppend('div', repoDetailsDiv, { class: 'divClass' });
+
+        createAndAppend('span', descriptionDiv, {
+          id: 'descSpan',
+          text: 'Description:',
+          class: 'spanClass',
+        });
+        createAndAppend('span', forkDiv, {
+          id: 'forkSpan',
+          text: 'Forks:',
+          class: 'spanClass',
+        });
+        createAndAppend('span', updatedDiv, {
+          id: 'updateSpan',
+          text: 'Updated:',
+          class: 'spanClass',
+        });
+        createAndAppend('span', repoDiv, { id: 'repoSpan', text: 'Repository:' });
+        createAndAppend('p', descriptionDiv, {
+          id: 'descP',
+          text: getDivContent('description', data, selectedOption),
+          class: 'pClass',
+        });
+        createAndAppend('p', forkDiv, {
+          id: 'forkP',
+          text: getDivContent('forks', data, selectedOption),
+          class: 'pClass',
+        });
+        createAndAppend('p', updatedDiv, {
+          id: 'updateP',
+          text: getDivContent('updated_at', data, selectedOption),
+          class: 'pClass',
+        });
+        createAndAppend('a', repoDiv, {
+          text: getDivContent('name', data, selectedOption),
+          id: 'repoA',
+          href: getDivContent('html_url', data, selectedOption),
+          target: '_blank',
+        });
+        const selectA = document.getElementById('selectId');
+        selectA.onchange = function func() {
+          selectedOption = document.getElementById('selectId').options[
+            document.getElementById('selectId').selectedIndex
+          ].text;
+          document.getElementById('repoA').href = getDivContent('html_url', data, selectedOption);
```

Comment:
Because you now have a variable `repoA` you no longer need this `getElementById`:

```js
repoA.href = selectedRepo.html_url;
repoA.innerText = selectedRepo.name;
descP.innerText = selectedRepo.description;
```

etc.